llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -761,6 +762,64 @@ static void updatePhysDepsDownwards(const MachineInstr *UseMI,
   }
 }
 
+/// Estimates the number of cycles elapsed between DefMI and UseMI if they're
+/// non-null and in the same BasicBlock. Returns std::nullopt when UseMI is in a
+/// different MBB than DefMI.
```

Comment:
This function is currently estimating the number of cycles elapsed between DefMi inclusive and UseMI exclusive. I think this is correct based on how this function is used, since its representing a latency between two instructions and latency is calculated from issue of def to when it is available to use.

nit: Do you think you could clarify the inclusive/exclusive relationship in this comment to make it clear?